src/web/app/components/response-time-series-chart/response-time-series-chart.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, Input, OnInit } from '@angular/core';
+import * as d3 from 'd3';
+import { ResponseTimeSeriesChartModel } from "./response-time-series-chart.model";
+import { FeedbackResponseStatsService } from "../../../services/feedback-response-stats.service";
+import { BehaviorSubject, interval } from "rxjs";
+import { switchMap, tap } from "rxjs/operators";
+import { FeedbackResponseStats } from "../../../types/api-output";
+import { ErrorMessageOutput } from "../../error-message-output";
+
+/**
+ * Response-tracking time series chart for admin user.
+ */
+@Component({
+  selector: 'tm-response-time-series-chart',
+  templateUrl: './response-time-series-chart.component.html',
+  styleUrls: ['./response-time-series-chart.component.scss']
+})
+export class ResponseTimeSeriesChartComponent implements OnInit {
+
+  @Input()
+  model: ResponseTimeSeriesChartModel = {
+    durationMinutes: 60,
+    intervalMinutes: 1
+  }
+
+  interval: BehaviorSubject<number> = new BehaviorSubject<number>(1);
+
+  constructor(private feedbackResponseStatsService: FeedbackResponseStatsService) { }
+
+  ngOnInit(): void {
+    this.interval.pipe(
+        switchMap(value => interval(value * 1000)), // in seconds (for easier testing), actual one will be in minutes
+        tap(() => this.refresh())
+    ).subscribe();
+  }
+
+  refresh() {
+    this.feedbackResponseStatsService.loadResponseStats(this.model.durationMinutes.toString(), this.model.intervalMinutes.toString())
+        .subscribe((data: FeedbackResponseStats) => {
+          console.log(data);
+          // parse data
+          const parsedData = this.parseData(data);
+          // draw chart
+          this.drawChart(parsedData, this.model.durationMinutes * 60 * 1000);
+        }, (resp: ErrorMessageOutput) => {
+          console.log((resp.error));
+        });
+  }
+
+  /**
+   * Handles a change in duration
+   */
+  setDurationHandler(duration: number): void {
+    this.model.durationMinutes = duration;
+  }
+
+  /**
+   * Handles a change in interval
+   */
+  setIntervalHandler(interval: number): void {
+    this.model.durationMinutes = interval;
+    this.interval.next(interval);
+  }
+
+  drawChart(data: object[], duration: number): void {
+
+    const svgWidth: number = 800;
+    const svgHeight: number = 400;
+    const margin = { top: 20, right: 20, bottom: 30, left: 50 };
+    const width: number = svgWidth - margin.left - margin.right;
+    const height: number = svgHeight - margin.top - margin.bottom;
+
+    var svg = d3.select('svg')
+        .attr("width", svgWidth)
+        .attr("height", svgHeight);
+
+    var g = svg.append("g")
```

Comment:
I think we should avoid having one-letter variables, or at least accompany them with some comments